ophyd/v2/epics.py

```diff
@@ -2,304 +2,113 @@
-
-    @property
-    def source(self) -> str:
-        return self.write_channel.source
-
-    async def connect(self, prefix: str = "", sim=False):
-        assert self.write_pv is not None, "Write PV not set"
-        self.write_channel = _make_channel_class(
-            prefix + self.write_pv, type(self.write_value), sim, self.write_channel
-        )
-        await self.write_channel.connect()
+try:
+    from ._p4p import PvaSignalBackend
+except ImportError as pva_error:
 
-    async def execute(self) -> None:
-        """Execute the Signal, putting `write_value` to `write_pv`"""
-        await self.write_channel.put(self.write_value, wait=self.wait)
+    def PvaSignalBackend(*args, pva_error=pva_error, **kwargs):  # type: ignore
+        raise NotImplementedError("PVA support not available") from pva_error
 
 
 class EpicsTransport(Enum):
     """The sorts of transport EPICS support"""
 
     #: Use Channel Access (using aioca library)
-    ca = ChannelCa
+    ca = CaSignalBackend
     #: Use PVAccess (using p4p library)
-    pva = ChannelCa  # TODO change to ChannelPva when Alan's written it
+    pva = PvaSignalBackend
 
 
 _default_epics_transport = EpicsTransport.ca
 
 
-def set_default_epics_transport(epics_transport: EpicsTransport):
-    """Set the default EPICS transport. Defaults to `EpicsTransport.ca`.
-
-    If PVs are specified as ca://PV or pva://PV they will be connected with the
-    requested transport. If unspecified the default EPICS transport will be
-    used.
-    """
-    global _default_epics_transport
-    _default_epics_transport = epics_transport
-
-
-def _make_channel_class(
-    pv: str,
-    datatype: Type[T],
-    sim=False,
-    existing_channel: Channel = DISCONNECTED_CHANNEL,
-) -> Channel[T]:
+def _transport_pv(pv: str) -> Tuple[EpicsTransport, str]:
     split = pv.split("://", 1)
     if len(split) > 1:
         # We got something like pva://mydevice, so use specified comms mode
-        transport, pv = split
-        pv_mode = EpicsTransport[transport]
+        transport_str, pv = split
+        transport = EpicsTransport[transport_str]
     else:
         # No comms mode specified, use the default
-        pv_mode = _default_epics_transport
-    if existing_channel is not DISCONNECTED_CHANNEL:
-        assert (
-            existing_channel.pv == pv
-        ), f"Reconnect asked to change from {existing_channel.pv} to {pv}"
-    if sim:
-        pv_cls = ChannelSim
-    else:
-        pv_cls = pv_mode.value
-    return pv_cls(pv, datatype)
+        transport = _default_epics_transport
+    return transport, pv
+
+
+def _make_backend(
+    datatype: Optional[Type[T]], read_pv: str, write_pv: str
+) -> SignalBackend[T]:
+    r_transport, r_pv = _transport_pv(read_pv)
+    w_transport, w_pv = _transport_pv(write_pv)
+    transport = get_unique({read_pv: r_transport, write_pv: w_transport}, "transports")
+    return transport.value(datatype, r_pv, w_pv)
```

Comment:
> Here you've allowed a backend to be made with different read/write transports. Is this practical? Will anyone ever want to read using channel access but get using pva for example? I know this should raise a TypeError if they're different, but this seems a bit difficult to debug/interpret if somehow this ever happens.

Technically this doesn't make a backend, it fails in `get_unique` before the backend is created with an error message saying the read and write pvs have different transports.  

> Is it worth for the factory functions (epics_signal_rw) adding a check making sure either ca/pva is specified in both read/write pvs? Maybe even a simple assert statement would do.

You could do that, but I think this will give a reasonable error message. Would you mind writing a test to check?

> Having said that I just don't know if you'd ever want a mixture of both.

No, we would never want a mixture of both